### PR TITLE
Some fixme in DDL service

### PR DESCRIPTION
**Pull request by hzhzhang**
add retry in append dwal block
add table exist check in rest

PR checklist:
- Did you run ClangFormat ? yes
- Did you run CheckStyle ? yes
- Did you check the comment / log / exception conventions in Engineering code process wiki page ? yes
- Did you import unncessary headers ? no
- Did you surround `Daisy : starts/ends` for new code in existing ClickHouse code base ? yes

Please write user-readable short description of the changes:

### CODE REVIEW

File: src/Interpreters/BlockUtils.cpp [hunk 1]
```diff
@@ -122,13 +127,25 @@ void appendBlock(Block && block, ContextPtr context, IDistributedWriteAheadLog::
...
+            LOG_INFO(log, "Successfully append record to DistributedWriteAheadLog, query_id={}", context->getCurrentQueryId());
+            return;
+        }
+
+        LOG_WARNING(
+            log,
+            "Failed to append record to DistributedWriteAheadLog, query_id={}, error={}, tried_times={}",
+            context->getCurrentQueryId(),
+            result_code,
+            i + 1);
```
Comment (chenziliang) [inline src/Interpreters/BlockUtils.cpp:145]: we shall sleep here ?

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp [hunk 2]
```diff
@@ -112,6 +114,13 @@ String TableRestRouterHandler::executeGet(const Poco::JSON::Object::Ptr & /* pay
 
 String TableRestRouterHandler::executePost(const Poco::JSON::Object::Ptr & payload, Int32 & /*http_status*/) const
 {
+    const auto & database_name = getPathParameter("database");
```
Comment (chenziliang) [inline src/Server/RestRouterHandlers/TableRestRouterHandler.cpp:117]: do we need update `http_status` in this function ?

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp [hunk 3]
```diff
@@ -152,14 +166,19 @@ String TableRestRouterHandler::executePatch(const Poco::JSON::Object::Ptr & payl
 
 String TableRestRouterHandler::executeDelete(const Poco::JSON::Object::Ptr & /*payload*/, Int32 & /*http_status*/) const
 {
+    const String & database_name = getPathParameter("database");
```
Comment (chenziliang) [inline src/Server/RestRouterHandlers/TableRestRouterHandler.cpp:169]: Update `https_status` ?

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp [hunk 4]
```diff
@@ -152,14 +166,19 @@ String TableRestRouterHandler::executePatch(const Poco::JSON::Object::Ptr & payl
...
+    const String & table_name = getPathParameter("table");
+
+    if (!tableExists(database_name, table_name))
+    {
+        return jsonErrorResponse(fmt::format("Table {}.{} doesn't exist", database_name, table_name), ErrorCodes::UNKNOWN_TABLE);
+    }
+
     if (query_context->isDistributed() && getQueryParameter("distributed_ddl") != "false")
```
Comment (chenziliang) [inline src/Server/RestRouterHandlers/TableRestRouterHandler.cpp:177]: Can we implement a `getPathParameter` function as below to simplify this checking and please update the code every place to switch to this new function when we need to bool checking

```
bool getPathParameterBool(const String & name, bool default_value=false) const
{
    const auto & value = getQueryParameter(name);
    if (value.empty())
    {
        return defalut_value;
    }
   
    if (value == "false" || value == "0")
    {
        return false;
    }
    return true;
}
```

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp [hunk 5]
```diff
@@ -326,4 +345,11 @@ String TableRestRouterHandler::buildResponse() const
     return resp_str_stream.str();
 }
 
+bool TableRestRouterHandler::tableExists(const String & database, const String & table) const
+{
+    const auto & catalog_service = CatalogService::instance(query_context);
```
Comment (chenziliang) [inline src/Server/RestRouterHandlers/TableRestRouterHandler.cpp:350]: Let's implement `tableExists` in CatalogService, in that way we don't need return the table definition

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp (hunk 2 above)
Comment (hzhzhang) [reply src/Server/RestRouterHandlers/TableRestRouterHandler.cpp:117]: fixed

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp (hunk 4 above)
Comment (hzhzhang) [reply src/Server/RestRouterHandlers/TableRestRouterHandler.cpp:177]: This will be fixed in PR #56

File: src/Interpreters/BlockUtils.cpp (hunk 1 above)
Comment (hzhzhang) [reply src/Interpreters/BlockUtils.cpp:145]: fixed

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp (hunk 5 above)
Comment (hzhzhang) [reply src/Server/RestRouterHandlers/TableRestRouterHandler.cpp:350]: fixed

File: src/Server/RestRouterHandlers/TableRestRouterHandler.cpp (hunk 3 above)
Comment (hzhzhang) [reply src/Server/RestRouterHandlers/TableRestRouterHandler.cpp:169]: fixed

File: src/Server/RestRouterHandlers/TableRestRouterHandler.h [hunk 6]
```diff
@@ -44,7 +47,7 @@ class TableRestRouterHandler : public RestRouterHandler
         const Poco::JSON::Object::Ptr & payload, const String & time_column, const String & default_order_by_granularity) const = 0;
 
 private:
-    bool tableExists(const String & database, const String & table) const;
+    CatalogService & catalog;
```
Comment (chenziliang) [inline src/Server/RestRouterHandlers/TableRestRouterHandler.h:50]: this is not actually quite good. Let's use the `instance()` function to get catalog

**Pull request merged**